### PR TITLE
rand: freebsd update, using getrandom.

### PR DESCRIPTION
**Pull request by devnexen**
supported since the 12th release, while 11.4 is EOL since 2021.

### CODE REVIEW

**Comment by sanxiyn** [comment]
While I agree Rust supporting FreeBSD versions unsupported by upstream is stupid, I think it should be an explicit decision, not like this. Note that while FreeBSD 11 was EOL since 2021, Rust CI continued to use FreeBSD 11 until [June 2022](https://github.com/rust-lang/rust/pull/97944).

**Comment by devnexen** [comment]
what would be your suggestion ?

**Comment by sanxiyn** [comment]
FCP on supported FreeBSD versions is now in progress at #89058.

**Comment by Amanieu** [comment]
FCP to drop FreeBSD 11 has completed (#89058).

@bors r+

**Comment by devnexen** [comment]
cc @Amanieu

File: library/std/src/sys/unix/rand.rs [hunk 1]
```diff
@@ -26,10 +25,10 @@ mod imp {
     use crate::fs::File;
     use crate::io::Read;
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "freebsd"))]
     use crate::sys::weak::syscall;
 
-    #[cfg(any(target_os = "linux", target_os = "android"))]
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "freebsd"))]
```
Comment (m-ou-se) [inline library/std/src/sys/unix/rand.rs:31]: `GRND_INSECURE` seems to be meaningless on FreeBSD. (It's non-existent on FreeBSD 12, and equivalent to `GRND_NONBLOCK` on FreeBSD 13+.) So re-using this same logic with `GRND_INSECURE_AVAILABLE` for Linux doesn't seem very useful.

Comment (devnexen) [reply library/std/src/sys/unix/rand.rs]: good point

**Comment by workingjubilee** [comment]
Seems like m-ou-se's comments were addressed!
@bors r+ rollup=always

**Pull request merged**